torch/fx/experimental/symbolic_shapes.py

```diff
@@ -269,20 +326,68 @@ def eval(cls, base, divisor):
     'sym_sqrt': lambda a: sympy.sqrt(a),
 }
 
+sizes_strides_methods = {
+    'is_non_overlapping_and_dense': lambda *args: IsNonOverlappingAndDenseIndicator(*args),
+}
+
+def eval_is_non_overlapping_and_dense(sizes, strides):
+    dim = len(sizes)
+
+    # Short-circuits for tensors of rank one, which are
+    # non-overlapping and "dense" if their stride is one
+    if dim == 1:
+        return strides[0] == 1
+
+    # Checks that there exists a permutation of the strides s.t. the tensor would be contiguous
+    # Sorts (length, stride) pairs by stride
+    lengths_and_strides = sorted(
+        tuple(zip(sizes, strides)), key=operator.itemgetter(1)
```

Comment:
Is the tuple necessary here? Can't sorted just run on the iterator form zip?

Reply:
preexisting condition from the copy pasted code